### ISSUE TITLE
Code details

### ISSUE THREAD

**Issue by fairy149** [emadeldeen24/AttnSleep#9]
Hello, thank you very much for sharing your code, and would like to ask you for the position of the related code mentioned in the paper on the class-aware loss function and optimization. Since I'm using a window environment, I can only train any one k fold alone during training

**Comment by emadeldeen24** [comment]
For the class-aware loss function, I first calculate the [samples count in each class](https://github.com/emadeldeen24/AttnSleep/blob/37eb7acf59bde9cb979a2665d5331277e0a60b69/data_loader/data_loaders.py#L45) in the data loader, then I use it with to [calculate the weights](https://github.com/emadeldeen24/AttnSleep/blob/37eb7acf59bde9cb979a2665d5331277e0a60b69/utils/util.py#L62), and finally, I use the [weighted cross-entropy loss](https://github.com/emadeldeen24/AttnSleep/blob/37eb7acf59bde9cb979a2665d5331277e0a60b69/model/loss.py#L5).

For the optimizer, it is defined [here](https://github.com/emadeldeen24/AttnSleep/blob/37eb7acf59bde9cb979a2665d5331277e0a60b69/train_Kfold_CV.py#L51), but you can change it from the [config file](https://github.com/emadeldeen24/AttnSleep/blob/37eb7acf59bde9cb979a2665d5331277e0a60b69/config.json#L16).

**Comment by fairy149** [comment]
Thank you very much for solving my puzzles, in fact I found the cross-entropy loss function under the loss.py file weighted_CrossEntropyLoss, can you calculate the sample count in each class in the data loader, use it to calculate the position of the code to calculate the weights for me, I did not find this part of the code, I will be grateful



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年8月18日(星期四) 晚上8:13
收件人: "emadeldeen24/AttnSleep";
抄送: "Author";
主题: Re: [emadeldeen24/AttnSleep] Code details (Issue #9)





 
For the class-aware loss function, I first calculate the samples count in each class in the data loader, then I use it with to calculate the weights, and finally, I use the weighted cross-entropy loss.
 
For the optimizer, it is defined here, but you can change it from the config file.
 
&mdash;
